Cross-compilation for arm64 fails

Build logs:

```
root@host:~/opentelemetry-ebpf-profiler# make agent TARGET_ARCH=arm64
docker run -v "$PWD":/agent -it --rm --user 0:0 profiling-agent \
   "make TARGET_ARCH=arm64 VERSION=v0.0.0 REVISION=main-aea5a171 BUILD_TIMESTAMP=1734118658"
go generate ./...
make  -C support/ebpf
make[1]: Entering directory '/agent/support/ebpf'
go run ../../tools/errors-codegen/main.go bpf errors.h
.
.
.
fork/exec /tmp/go-build2645483506/b001/exe/main: exec format error
metrics/types.go:7: running "go": exit status 1
make: *** [Makefile:69: generate] Error 1
make: *** Waiting for unfinished jobs....
fork/exec /tmp/go-build1559362405/b001/exe/main: exec format error
make[1]: *** [Makefile:72: errors.h] Error 1
make[1]: Leaving directory '/agent/support/ebpf'
make: *** [Makefile:72: ebpf] Error 2
make: *** [Makefile:122: agent] Error 2
```

Using `debian:latest` to build docker-image to get around the clang-16 not available issue.

--- a/Makefile
+++ b/Makefile
@@ -66,7 +66,7 @@ clean:
 	@rm -rf go .cache
 
 generate:
-	go generate ./...
+	GOARCH=$(NATIVE_ARCH) go generate ./...
 
 ebpf:
 	$(MAKE) $(EBPF_FLAGS) -C support/ebpf

